scripts/benchmarks/weave_setup_time.py

```diff
@@ -0,0 +1,322 @@
+
+    # Calculate stats for each timing type
+    stats = {}
+    for timing_type, times in results.items():
+        stats[timing_type] = calculate_stats(times)
+
+    with open(filename, "w", newline="") as csvfile:
+        writer = csv.writer(csvfile)
+
+        # Write header
+        writer.writerow(["Timing_Type", "Metric", "Value_Seconds"])
+
+        # Write data rows
+        metrics = ["mean", "median", "std_dev", "min", "max"]
+        metric_labels = ["Mean", "Median", "Std Dev", "Min", "Max"]
+
+        for timing_type in ["import", "init", "total"]:
+            for metric, label in zip(metrics, metric_labels):
+                value = stats[timing_type][metric]
+                writer.writerow([timing_type.title(), label, f"{value:.6f}"])
+
+
+def read_results_from_csv(filename: str) -> list[dict[str, str]]:
+    """Read benchmark results from CSV file.
+
+    Args:
+        filename: Input CSV filename.
+
+    Returns:
+        list[dict[str, str]]: List of row dictionaries.
+    """
+    if not os.path.exists(filename):
+        console.print(f"[red]Results file {filename} not found[/red]")
+        return []
+
+    console.print(f"Reading results from {filename}...")
+
+    results = []
+    with open(filename, newline="") as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            results.append(row)
+
+    return results
+
+
+def create_results_table_from_csv(csv_data: list[dict[str, str]]) -> Table:
+    """Create a Rich table from CSV data.
+
+    Args:
+        csv_data: List of row dictionaries from CSV.
+
+    Returns:
+        Table: Rich table with timing results.
+    """
+    table = Table(title="Weave Setup Timing Results")
+    table.add_column("Component", style="cyan", no_wrap=True)
+    table.add_column("Metric", style="magenta", no_wrap=True)
+    table.add_column("Time (seconds)", style="green", justify="right")
+
+    for row in csv_data:
+        timing_type = row["Timing_Type"]
+        metric = row["Metric"]
+        value = float(row["Value_Seconds"])
+
+        table.add_row(timing_type, metric, f"{value:.3f}s")
+
+    return table
+
+
+def main():
+    """Benchmark weave setup time and log results."""
+    parser = argparse.ArgumentParser(description="Benchmark Weave setup time")
+    parser.add_argument(
+        "--iterations",
+        type=int,
+        default=10,
+        help="Number of setup iterations to run (default: 10)",
+    )
+
+    args = parser.parse_args()
+
+    console.print("[bold]Weave Setup Time Benchmark[/bold]")
+    console.print(f"Running {args.iterations} iterations...\n")
+
+    # Run the benchmark
+    results = run_benchmark_iterations(args.iterations)
+
+    # Write results to CSV in the same directory as the script
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+    csv_filename = os.path.join(script_dir, "weave_setup_time_results.csv")
+    write_results_to_csv(results, csv_filename)
```

Comment:
Might want to make `-w` `--write` a cli arg with a default of false which can be a boolean or a string value. if string, name the file the input, if set with no name, default to the existing one.